tools/markdown-magic/src/transforms/includeCodeTransform.cjs

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const fs = require("fs");
+const {
+	formattedEmbeddedContentBody,
+	formattedSectionText,
+	readFile,
+	resolveRelativePath,
+} = require("../utilities.cjs");
+
+/**
+ * Embeds contents from the specified file paths within the provided (optional) line boundaries.
+ *
+ * @param {object} content - The original document file contents.
+ * @param {object} options - Transform options.
+ * @param {string} options.path - Relative path from the document to the file being embedded.
+ * @param {string | undefined} options.start - (optional) First line from the target file to be embedded (inclusive).
+ * Expected to be a string-formatted integer.
+ * Default: 0.
+ * Constraints are the same as those for the `end` parameter to
+ * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#parameters | Array.slice}
+ * @param {string | undefined} options.end - (optional) Line of the target file at which to end the embedded range (exclusive).
+ * Expected to be a string-formatted integer.
+ * Default: <file-line-count> + 1.
+ * Constraints are the same as those for the `end` parameter to
+ * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#parameters | Array.slice}
+ * @param {string | undefined} options.language - The code language to use for syntax highlighting.
+ * E.g., "language=typescript" would yield a markdown codeblock starting with "\`\`\`typescript".
+ * Default: No language specified.
+ * @param {object} config - Transform configuration.
```

Comment:
Just curious, specifying the whole `config` in addition to the sub-properties works as expected? Even with this one in between some of its sub-properties?

Reply:
Can you elaborate? I'm not sure I understand the question.

Reply:
Oh, in terms of the jsdoc comments? Yes, that is supported, and intellisense displays both.

Reply:
My bad, this is `config`, not `options` like above so there's no funky ordering. I guess my question would still be "do we need to specify `config` and `config.originalPath` separately" and if Intellisense does the right thing, but I imagine it does :).

Reply:
We don't strictly need the docs for `config`, the separate params are probably sufficient. It just felt incomplete to me without them.

Intellisense looks like this when hovering over the config:
![image](https://github.com/user-attachments/assets/c689c400-2798-47d1-9444-01f4bee3f88b)